### ISSUE TITLE
Since 18 August distribution of posts no longer works

### ISSUE THREAD

**Issue by copiis** [friendica/friendica#7584]
Since the git pull on 18 August, the distribution of posts no longer works. Neither to Buffer, Wordpress, Twitter or Tumbrl.

**Comment by nupplaphil** [comment]
Could that be related to the lock issue you found? If so, maybe you can check if it would work with lockprovider 'database'

**Comment by annando** [comment]
Do you have any errors in your logs?

**Comment by copiis** [comment]
My log is full of timeout messages like these.
Have only such timeout messages, the whole log is full.

`[Thu Aug 29 17:28:55.229275 2019] [proxy_fcgi:error] [pid 8709:tid 140492555310848] (70007)The timeout specified has expired: [client 88.99.89.29:51846] AH01075: Error dispatching request to : (polling)
`
After I switched to the new composer on the 18th, I got this message and then only the timeout messages.
`[Sun Aug 18 22:43:56.854252 2019] [proxy_fcgi:error] [pid 17353:tid 139766905825024] [client 89.10.144.27:54592] AH01071: Got error 'PHP message: PHP Notice:  Undefined offset: 401 in /addon/buffer/bufferapp.php on line 134\n', referer: https://meld.de/settings
[Sun Aug 18 22:49:05.197378 2019] [proxy_fcgi:error] [pid 17353:tid 139766847076096] [client 89.10.144.27:54878] AH01071: Got error 'PHP message: PHP Notice:  Undefined offset: 401 in /addon/buffer/bufferapp.php on line 134\n', referer: https://meld.de/settings
[Sun Aug 18 22:53:14.754463 2019] [proxy_fcgi:error] [pid 17353:tid 139766637135616] [client 89.10.144.27:55016] AH01071: Got error 'PHP message: PHP Notice:  Undefined offset: 401 in /addon/buffer/bufferapp.php on line 134\n', referer: https://meld.de/settings/addon
[Sun Aug 18 22:57:59.405752 2019] [proxy_fcgi:error] [pid 4603:tid 139766305625856] (70007)The timeout specified has expired: [client 163.172.7.58:40136] AH01075: Error dispatching request to : (polling)
[Sun Aug 18 22:58:24.965513 2019] [proxy_fcgi:error] [pid 4603:tid 139766595057408] (70007)The timeout specified has expired: [client 163.172.7.58:42294] AH01075: Error dispatching request to : (polling)
[Sun Aug 18 23:08:55.803380 2019] [proxy_fcgi:error] [pid 8756:tid 139766679140096] [client 89.10.144.27:55682] AH01071: Got error 'PHP message: PHP Notice:  Undefined offset: 401 in /addon/buffer/bufferapp.php on line 134\n', referer: https://meld.de/settings
[Sun Aug 18 23:09:08.625606 2019] [proxy_fcgi:error] [pid 17353:tid 139766712743680] [client 89.10.144.27:55688] AH01071: Got error 'PHP message: PHP Notice:  Undefined offset: 401 in /addon/buffer/bufferapp.php on line 134\n', referer: https://meld.de/settings/connectors
[Sun Aug 18 23:10:37.096371 2019] [proxy_fcgi:error] [pid 8756:tid 139766679140096] [client 89.10.144.27:55696] AH01071: Got error 'PHP message: PHP Notice:  session_start(): A session had already been started - ignoring in /addon/buffer/buffer.php on line 105\nPHP message: PHP Notice:  Undefined index: oauth in /addon/buffer/bufferapp.php on line 126\n', referer: https://meld.de/settings/connectors
[Sun Aug 18 23:10:59.032917 2019] [proxy_fcgi:error] [pid 8756:tid 139766679140096] [client 89.10.144.27:55716] AH01071: Got error 'PHP message: PHP Notice:  session_start(): A session had already been started - ignoring in /addon/buffer/buffer.php on line 105\n', referer: https://buffer.com/oauth2/authorize?client_id=5d4eb476b99b1f4105202644&redirect_uri=https%3A%2F%2Fmeld.de%2Fbuffer%2Fconnect&response_type=code`

**Comment by MrPetovan** [comment]
I can't confirm, distribution to Twitter worked for me 20 minutes ago on latest RC.

**Comment by copiis** [comment]
I receive posts from Twitter and can comment on them, only Friendica's own are not forwarded to Twitter.
I've now disabled that, on my Twitter profile on Friendica, the posts are mirrored as my own, maybe that's the mistake.

**Comment by MrPetovan** [comment]
That would indeed prevent your Friendica posts to be forwarded to Twitter. It is the sole intent and purpose of this setting switch.

**Comment by copiis** [comment]
Ok on Tumblr i tryed to reautheticate but i get a 

> Could not connect to Tumblr. Refresh the page or try again later.

**Comment by copiis** [comment]
> Could that be related to the lock issue you found?

What do you mean?

**Comment by copiis** [comment]
> That would indeed prevent your Friendica posts to be forwarded to Twitter.

After switching it still not works.

**Comment by copiis** [comment]
Aren't the checkboxes missing to confirm that the contribution will be forwarded to the other social media or not?

![MELD - Free Speech  No Advertising  No Spyware  _ Compose - Edited](https://user-images.githubusercontent.com/12480023/64074990-aa9a6180-ccb2-11e9-8fc0-0208c97e7e09.gif)

**Comment by MrPetovan** [comment]
Indeed, it seems to show the addons aren't correctly enabled/configured. For example, if the Twitter addon is correctly enabled and your personal Twitter connector setting "Allow posting to Twitter" is enabled, the checkbox should show.

**Comment by copiis** [comment]
Everything seems to be right in the settings.
![2019-09-01 14 07 06](https://user-images.githubusercontent.com/12480023/64076264-e89f8180-ccc2-11e9-9358-41fb0596a972.png)

**Comment by annando** [comment]
Have you checked what happens when you use the "old" message creation dialog?

**Comment by AlfredSK** [comment]
Just for the record. I'm using the Twitter add-on too and it shows up on the new compose post page.

**Comment by copiis** [comment]
> Have you checked what happens when you use the "old" message creation dialog?

Yes, the connectors are there but nothing is distributed.
![2019-09-04 17 55 57](https://user-images.githubusercontent.com/12480023/64271380-81413600-cf3d-11e9-8cd2-31da272af5dc.png)

**Comment by copiis** [comment]
> Just for the record. I'm using the Twitter add-on too and it shows up on the new compose post page.

And the post are distributed?

**Comment by AlfredSK** [comment]
Yes. I checked it some hours ago. But I don't cross-post to Twitter by default.

**Comment by AlfredSK** [comment]
What I see in that log snippet above is that there seems to be an issue with the buffer add-on. I would try to disable it and have a look if something is changing.

**Comment by copiis** [comment]
Again in detail. With me these plugins are set in such a way that they post by default.
With me the Wordpress Addon does not work. Last post to https://copiis.home.blog on August 18th.
In the admin area I de- and re-activated the plugin, and re-entered the access data under Settings.

On August 18th also the Twitter plugin stopped sending to Twitter. The direction Twitter Friendica works, but the direction Friendica Twitter does not. In the settings I disconnected the connection to Twitter and re-established it.

Tumbrl the same, when I try to authenticate the plugin under Settings I get the following message.

`Could not connect to Tumblr. Refresh the page or try again later.`

and in the log this is written here.

`PHP Notice: session_start(): A session had already been started - ignoring in /home/users/meld/www/friendica/addon/tumblr/tumblr.php on line 101\nPHP message: PHP Notice:  Undefined index: oauth_token in /home/users/meld/www/friendica/addon/tumblr/library/tumblroauth.php on line 79\nPHP message: PHP Notice:  Undefined index: oauth_token_secret in /home/users/meld/www/friendica/addon/tumblr/library/tumblroauth.php on line 79\nPHP message: PHP Notice:  Undefined index: oauth_token in /home/users/meld/www/friendica/addon/tumblr/tumblr.php on line 125\nPHP message: PHP Notice:  Undefined index: oauth_token_secret in /home/users/meld/www/friendica/addon/tumblr/tumblr.php on line 126\n', referer: https://meld.de/settings/connectors`

I'm posting on LinkeIt via Buffer. The Buffer addon can be reauthenticated but there hasn't been a post on LinkedIn since August 18th.

On August 18th I switched to the new composer page, maybe it has something to do with that?

And my log is full with timeout messages since weeks 

`[Thu Sep 05 06:09:22.370402 2019] [proxy_fcgi:error] [pid 31231:tid 140074289891072] (70007)The timeout specified has expired: [client 163.172.7.58:46324] AH01075: Error dispatching request to : (polling)`

**Comment by copiis** [comment]
> That would indeed prevent your Friendica posts to be forwarded to Twitter. It is the sole intent and purpose of this setting switch.

I just sent a post from Twitter and it was published on the Friendica site as self_remot although I changed that for my Twitter contact.
![Bildschirmfoto vom 2019-09-09 17-56-50](https://user-images.githubusercontent.com/12480023/64546788-98ac6480-d32b-11e9-9431-42fc41afa424.png)

**Comment by copiis** [comment]
The error has occurred since I set my Twitter profile that was present in my Friendica contact list as remote_self. 
And although I undone this change, all posts I made in Friendica were marked as coming from Twitter.
@annando  @MrPetovan  do you have a hint in which direction I should do the troubleshooting?
Strangely enough, this change has disrupted all connectors. To Wordpress, Tumbrl and Buffer, also nothing is sent anymore although they are still the valid settings.
Be Twitter gets this error message...
![2019-09-11 04 04 02](https://user-images.githubusercontent.com/12480023/64663821-4c594580-d44d-11e9-85ea-c0d090452c9c.png)

And on Tumbrl i get these when I try to (re-)authenticate the Tumbrl page
![2019-09-11 04 34 12](https://user-images.githubusercontent.com/12480023/64663860-73177c00-d44d-11e9-9e1f-b306ace8cf29.png)

**Comment by annando** [comment]
I don't think that it is related. I have to do some tests, but had been busy so far.

**Comment by copiis** [comment]
Could it be a change in the local.config.php that the connectors don't work anymore?

**Comment by annando** [comment]
Question is: Have you changed something? Until now no one reported the same problem. So it is likely that it is a local problem. Nonetheless we should solve it.

**Comment by copiis** [comment]
> Question is: Have you changed something?

On that day I just switched to the new Composer page and wrote from Twitter as remote_self.
None of my users has reported that the distribution doesn't work, so I think it's a problem with my account.
As an administrator you can do some damage, but that this is also possible as a user, that amazes me.

**Comment by copiis** [comment]
I think posting from Twitter as self_remote has done something with my account. I then deleted all posts I posted from Twitter to Friendica in the hope that the bug would be fixed. This restored the contact to Wordpress, Buffer and Tumbrl but the contact to Twitter remains dead.

For testing I have created a new account and posting to Twitter, Buffer and Wordpress works. So the problem only exists for my account.

If you want to authenticate or re-authenticate an account at Tumbrl you get the message "Could not connect to Tumblr. Refresh the page or try again later." Can anyone confirm this?

**Comment by copiis** [comment]
@annando do you have a tip where to look for a clue?
There must probably be something in the database, right?

**Comment by annando** [comment]
There is no easy answer, sorry. Currently I don't know where to look at.

**Comment by copiis** [comment]
So I still suspect that when I turned the `remot_self` on and off for the Twitter contact, the setting wasn't turned off again. Since then posting to Twitter didn't work anymore.
After that I deleted my Twitter contacts but they were not imported again.

Would it be possible for someone to reproduce this?

1. Enable remote_self for his Twitter contact.
2. Write a few posts on Twitter that show up on Friendica.
3. Reply from Friendica to your own posts on Twitter.
4. Delete the remote_self for the Twitter contact.
5. Delete the Twitter contact yourself.

Then watch if the own Twitter profile is imported again and if in the settings of the Twitter connector `Invalid Twitter info` is written and the `Disconect` button is missing.

Would it be possible to switch `remote_self` on or off in the database?

**Comment by copiis** [comment]
> There is no easy answer, sorry. Currently I don't know where to look at.

Now I created a new account with the email of the administrator of the site where I also have administration rights.
Could I delete the first admin account now?

**Comment by annando** [comment]
It's better to keep the old account.
